lib/hsa/CMakeLists.txt

```diff
@@ -1,9 +1,13 @@
+include(FindPkgConfig)
+
 ####################
 # HCC runtime (HSA implementation)
 ####################
 if (HAS_ROCM EQUAL 1)
 add_mcwamp_library_hsa(mcwamp_hsa mcwamp_hsa.cpp unpinned_copy_engine.cpp)
-target_link_libraries(mcwamp_hsa  unwind)
+pkg_check_modules(LIBUNWIND REQUIRED libunwind)
+# TODO: we should probably add required CFLAGS and LDFLAGS from pkgconfig
+target_link_libraries(mcwamp_hsa ${LIBUNWIND_LIBRARIES})
```

Comment:
> what's the advantage of imported target if we're not using any of the IMPORTED_ properties?

The downstream libraries use the imported targets. The advantage is the paths to the library are not hard coded so it allows everything to be relocatable.

> If I try to include the resulting hcc-config.cmake just after the call to configure_package_config_file it fails even without any changes because it uses set_and_check on install directories which don't exists at configure time (it also tries to check for /home/include which looks wrong)

This is to be used after installation, so users can do `find_package(hcc)` and then the user can use the hcc cmake targets(like `hccrt` or `hc_am`) directly in their project.